File: Linkpoint/src/main/java/com/linkpoint/slproto/users/manager/CurrentLocationInfo.java
package com.linkpoint.slproto.users.manager;

import com.linkpoint.slproto.users.ParcelData;
import com.linkpoint.voice.common.model.VoiceChannelInfo;
import javax.annotation.Nullable;

public abstract class CurrentLocationInfo {
    public static CurrentLocationInfo create(@Nullable ParcelData parcelData, int i, int i2, @Nullable VoiceChannelInfo voiceChannelInfo) {
        return new AutoValue_CurrentLocationInfo(parcelData, i, i2, voiceChannelInfo);
    }

    public abstract int inChatRangeUsers();

    public abstract int nearbyUsers();

    @Nullable
    public abstract ParcelData parcelData();

    @Nullable
    public abstract VoiceChannelInfo parcelVoiceChannel();
}

File: Linkpoint/src/main/java/com/linkpoint/slproto/users/manager/CurrentLocationInfo.kt
package com.linkpoint.slproto.users.manager

import com.linkpoint.slproto.users.ParcelData
import com.linkpoint.voice.common.model.VoiceChannelInfo

abstract class CurrentLocationInfo {
    abstract fun inChatRangeUsers(): Int
    abstract fun nearbyUsers(): Int
    abstract fun parcelData(): ParcelData?
    abstract fun parcelVoiceChannel(): VoiceChannelInfo?

    companion object {
        @JvmStatic
        fun create(
            parcelData: ParcelData?,
            nearbyUsers: Int,
            inChatRangeUsers: Int,
            voiceChannelInfo: VoiceChannelInfo?
        ): CurrentLocationInfo {
            return AutoValue_CurrentLocationInfo(parcelData, nearbyUsers, inChatRangeUsers, voiceChannelInfo)
        }
    }
}